src/compiler/propagation/type_propagator.cc

```diff
@@ -294,6 +294,17 @@ void TypePropagator::call_virtual(MethodTemplate* caller, TypeStack* stack, uint
   stack->drop_arguments(arity);
 }
 
+void TypePropagator::propagate_through_lambda(Method method) {
+  ASSERT(method.is_lambda_method());
+  std::vector<ConcreteType> arguments;
+  // TODO(kasper): Can we at least push an instance of the Lambda class
+  // as the receiver type?
+  for (int i = 0; i < method.arity(); i++) {
+    arguments.push_back(ConcreteType::any());
```

Comment:
Why can't we use the types that are on the stack?
Add a comment.

Reply:
We're instantiating -- not invoking -- the lambda here. Added comment.